packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -445,6 +445,19 @@ class EditableTextState extends State<EditableText> implements TextInputClient {
     _scrollController.jumpTo(_getScrollOffsetForCaret(caretRect));
   }
 
+  bool _textChangedSinceLastCaretUpdate = false;
+
+  void _handleCaretChanged(Rect caretRect) {
+    // If the caret location has changed due to an update to the text or
+    // selection, then scroll the caret into view.
+    if (_textChangedSinceLastCaretUpdate) {
+      _textChangedSinceLastCaretUpdate = false;
+      scheduleMicrotask(() {
+        _scrollController.jumpTo(_getScrollOffsetForCaret(caretRect));
```

Comment:
Can we make this an animateTo() ?

If we make it jumpTo(), then we're going to have a one-frame delay where we'll always first show the text field overflowing, then we'll shift it into view. If we use animateTo(), then we'll be smoothly scrolling the text field the whole time.